sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/tests/PlaywrightReporterTests.cs

```diff
@@ -0,0 +1,266 @@
+
+            environmentMock.Setup(e => e.Exit(It.IsAny<int>())).Callback<int>(i => { });
+            consoleWriterMock.Setup(c => c.WriteError(It.IsAny<string>())).Verifiable();
+
+            var xmlSettings = @"
+<RunSettings>
+    <TestRunParameters>
+        <Parameter name=""ServiceAuthType"" value=""No-auth"" />
+    </TestRunParameters>
+</RunSettings>
+";
+            var reporter = new PlaywrightReporter(logger: null, environment: environmentMock.Object, xmlRunSettings: null, consoleWriter: consoleWriterMock.Object, jsonWebTokenHandler: null);
+            reporter.InitializePlaywrightReporter(xmlSettings);
+
+            consoleWriterMock.Verify(c => c.WriteError(It.IsRegex("Failed to initialize PlaywrightServiceSettings")), Times.Once);
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_WithNoServiceUrl_PrintsMessageAndExits()
+        {
+            var consoleWriterMock = new Mock<IConsoleWriter>();
+            var environmentMock = new Mock<IEnvironment>();
+
+            environmentMock.Setup(e => e.Exit(It.IsAny<int>())).Callback<int>(i => { });
+            consoleWriterMock.Setup(c => c.WriteError(It.IsAny<string>())).Verifiable();
+
+            var reporter = new PlaywrightReporter(logger: null, environment: environmentMock.Object, xmlRunSettings: null, consoleWriter: consoleWriterMock.Object, jsonWebTokenHandler: null);
+
+            reporter.InitializePlaywrightReporter("");
+
+            consoleWriterMock.Verify(c => c.WriteError(Constants.s_no_service_endpoint_error_message), Times.Once);
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_WithNoAccessToken_PrintsMessageAndExits()
+        {
+            var consoleWriterMock = new Mock<IConsoleWriter>();
+            var environmentMock = new Mock<IEnvironment>();
+
+            environmentMock.Setup(e => e.Exit(It.IsAny<int>())).Callback<int>(i => { });
+            environmentMock.Setup(e => e.GetEnvironmentVariable(ReporterConstants.s_pLAYWRIGHT_SERVICE_REPORTING_URL)).Returns("reporting-url");
+            consoleWriterMock.Setup(c => c.WriteError(It.IsAny<string>())).Verifiable();
+
+            var reporter = new PlaywrightReporter(logger: null, environment: environmentMock.Object, xmlRunSettings: null, consoleWriter: consoleWriterMock.Object, jsonWebTokenHandler: null);
+
+            reporter.InitializePlaywrightReporter("");
+
+            consoleWriterMock.Verify(c => c.WriteError(Constants.s_no_auth_error), Times.Once);
+            environmentMock.Verify(e => e.Exit(1), Times.Once);
+            Assert.Null(reporter._testProcessor);
+        }
+
+        [Test]
+        public void InitializePlaywrightReporter_Default_SetsUpTestProcessor()
```

Comment:
should we add test for AccessToken flow ?

Reply:
Access Token does not have a different flow. There are utility functions to differentiate access token and entra token and get the token claims. Those functions have separate unit tests.